src/foundry/client/pixi/placeables/primary-canvas-objects/primary-sprite-mesh.d.mts

```diff
@@ -0,0 +1,110 @@
+import type { InexactPartial } from "../../../../../types/utils.d.mts";
+
+// TODO: Remove when #2570 is completed
+declare const PrimaryBaseSamplerShader: BaseSamplerShader;
+
+// TODO: Remove when #2878 is completed
+type TextureAlphaData = unknown;
+
+declare global {
+  /**
+   * A basic PCO sprite mesh which is handling occlusion and depth.
+   */
+  class PrimarySpriteMesh extends PrimaryOccludableObjectMixin(SpriteMesh) {
+    constructor(
+      options?:
+        | InexactPartial<{
+            /** Texture passed to the SpriteMesh. */
+            texture: PIXI.Texture;
+
+            /** The shader class used to render this sprite. */
+            shaderClass: typeof PrimaryBaseSamplerShader;
+
+            /** The name of this sprite. */
+            name: string;
+          }>
+        | PIXI.Texture,
+      shaderClass?: typeof PrimaryBaseSamplerShader,
```

Comment:
Looks like `PrimaryBaseSamplerShader.AnyConstructor` will do because it never does `new shaderClass` directly, only calls `shaderClass.create`.

I'm aware `PrimaryBaseSamplerShader.AnyConstructor` isn't created yet, you can stub it out with `namespace PrimaryBaseSamplerShader { type AnyConstructor = unknown }` for the time being.